File: file1_new.java
public class file1{
    private int day;
    private int month;
    private int year;
    file1_old(int day, int month, int year) {
        this.day = day;
        this.month = month;
        this.year = year;
    }
    public file1_old(file1_old p2) {
        this.day = p2.day;
        this.month = p2.month;
        this.year = p2.year;
    }
    private static int monthToNum(String m) {
        switch (m) {
            case "Jan": case "jan": return 1;
            case "Feb": case "feb": return 2;
            case "Mar": case "mar": return 3;
            case "Apr": case "apr": return 4;
            case "May": case "may": return 5;
            case "Jun": case "jun": return 6;
            case "Jul": case "jul": return 7;
            case "Aug": case "aug": return 8;
            case "Sep": case "sep": return 9;
            case "Oct": case "oct": return 10;
            case "Nov": case "nov": return 11;
            case "Dec": case "dec": return 12;
            default: return 0;
        }
    }
}

File: File1_New.py
class File1:
    def __init__(self, day: int, month: int, year: int):
        self.day = day
        self.month = month
        self.year = year
    @classmethod
    def from_file1(cls, other: "File1") -> "File1":
        """Copy constructor equivalent."""
        return cls(other.day, other.month, other.year)
    @staticmethod
    def month_to_num(m: str) -> int:
        """Convert month abbreviation to number.
        Returns 0 for unknown values."""
        if not m:
            return 0
        m_lower = m.lower()
        if m_lower in ("jan",):
            return 1
        if m_lower in ("feb",):
            return 2
        if m_lower in ("mar",):
            return 3
        if m_lower in ("apr",):
            return 4
        if m_lower in ("may",):
            return 5
        if m_lower in ("jun",):
            return 6
        if m_lower in ("jul",):
            return 7
        if m_lower in ("aug",):
            return 8
        if m_lower in ("sep",):
            return 9
        if m_lower in ("oct",):
            return 10
        if m_lower in ("nov",):
            return 11
        if m_lower in ("dec",):
            return 12
        return 0
